USBHelperLauncher/Net/ApplicationEndpoint.cs

```diff
@@ -29,26 +30,50 @@ public void GetZipProxy(Session oS)
             var data = GetRequestData(oS);
             byte[] bytes = Convert.FromBase64String(data.Get("url"));
             string url = Encoding.UTF8.GetString(bytes);
-            string content;
-            using (WebClient client = new WebClient())
+
+            try
             {
-                client.Proxy = Program.GetProxy().GetWebProxy();
-                content = client.DownloadString(url);
+                string content;
+                using (WebClient client = new WebClient())
+                {
+                    client.Proxy = Program.GetProxy().GetWebProxy();
+                    byte[] responseBytes = client.DownloadData(url);
+                    var contentType = new ContentType(client.ResponseHeaders["Content-Type"]);
+                    content = Encoding.GetEncoding(contentType.CharSet ?? "UTF-8").GetString(responseBytes);
+                }
+
+                MemoryStream stream = new MemoryStream();
+                using (ZipArchive zip = new ZipArchive(stream, ZipArchiveMode.Create))
+                {
+                    var entry = zip.CreateEntry("content");
+                    using (var streamWriter = new StreamWriter(entry.Open()))
+                    {
+                        streamWriter.Write(content);
+                    }
+                }
+
+                byte[] dataBytes = stream.ToArray();
+                oS.utilCreateResponseAndBypassServer();
+                oS.oResponse["Content-Length"] = dataBytes.Length.ToString();
+                oS.responseBodyBytes = dataBytes;
+                Proxy.LogRequest(oS, this, "Created zip from " + url);
             }
-            MemoryStream stream = new MemoryStream();
-            using (ZipArchive zip = new ZipArchive(stream, ZipArchiveMode.Create))
+            catch (WebException webEx)
             {
-                var entry = zip.CreateEntry("content");
-                using (var streamWriter = new StreamWriter(entry.Open()))
+                Proxy.LogRequest(oS, this, "Unable to create zip for " + url + ": " + webEx);
+
+                oS.utilCreateResponseAndBypassServer();
+                oS.utilSetResponseBody("");
```

Comment:
Not needed, `utilCreateResponseAndBypassServer();` already sets an empty response.